File: sample/dracula.scala
final case class Vampire(location: String, birthDate: Int, deathDate: Int, weaknesses: List[String]) {
  def age: Int = deathDate - birthDate
}

@main def runDraculaSample(): Unit = {
  val dracula = Vampire("Transylvania", 1428, 1476, List("Sunlight", "Garlic"))
  val alucard = Vampire("Wallachia", 1458, 1510, List("Holy Water"))

  println(s"Dracula: ${dracula.age}, Alucard: ${alucard.age}")
}

File: sample/dracula.rs
#[derive(Debug)]
struct Vampire {
    location: String,
    birth_date: u16,
    death_date: u16,
    weaknesses: Vec<String>,
}

impl Vampire {
    fn age(&self) -> u16 {
        self.death_date - self.birth_date
    }
}

fn main() {
    let dracula = Vampire {
        location: "Transylvania".to_owned(),
        birth_date: 1428,
        death_date: 1476,
        weaknesses: vec!["Sunlight".to_owned(), "Garlic".to_owned()],
    };

    let alucard = Vampire {
        location: "Wallachia".to_owned(),
        birth_date: 1458,
        death_date: 1510,
        weaknesses: vec!["Holy Water".to_owned()],
    };

    println!("Dracula: {}, Alucard: {}", dracula.age(), alucard.age());
}
